kore/test/Test/Kore/Reachability/Prove.hs

```diff
@@ -210,6 +210,48 @@ test_proveClaims =
          in [ mkTest "OnePath" simpleOnePathClaim
             , mkTest "AllPath" simpleAllPathClaim
             ]
+    , testGroup "returns first two stuck claims after disjunction" $
+        let axioms = [simpleAxiom Mock.a (mkOr Mock.b Mock.c `mkOr` Mock.d)]
+            mkTest name mkSimpleClaim =
+                testCase name $ do
+                    actual <-
+                        proveClaimsMaxCounterexamples_
+                            Unlimited
+                            (Limit 1)
+                            2
+                            axioms
+                            [mkSimpleClaim Mock.a Mock.e]
+                            []
+                    let stuck =
+                            [ mkSimpleClaim Mock.b Mock.e
+                            , mkSimpleClaim Mock.c Mock.e
+                            ]
+                        expect = MultiAnd.make (fmap StuckClaim stuck)
+                    assertEqual "" expect actual
+         in [ mkTest "OnePath" simpleOnePathClaim
+            , mkTest "AllPath" simpleAllPathClaim
+            ]
+    , testGroup "returns both stuck claims after disjunction" $
+        let axioms = [simpleAxiom Mock.a (mkOr Mock.b Mock.c)]
+            mkTest name mkSimpleClaim =
+                testCase name $ do
+                    actual <-
+                        proveClaimsMaxCounterexamples_
+                            Unlimited
+                            (Limit 1)
+                            3
+                            axioms
+                            [mkSimpleClaim Mock.a Mock.d]
+                            []
+                    let stuck =
+                            [ mkSimpleClaim Mock.b Mock.d
+                            , mkSimpleClaim Mock.c Mock.d
+                            ]
+                        expect = MultiAnd.make (fmap StuckClaim stuck)
+                    assertEqual "" expect actual
+         in [ mkTest "OnePath" simpleOnePathClaim
+            , mkTest "AllPath" simpleAllPathClaim
+            ]
```

Comment:
It might be redundant, but I think it's best to have an explicit test for the default case as well, it shows our intent and serves as documentation.